### PR TITLE
Introduce mapping layer for variant analysis history items

### PR DESCRIPTION
**Pull request by charisk**
Adds functions for DTO <-> domain model mappings. As part of this we've also modelled enums as DTOs which were previously not modelled.

Note that the mapping functions for the local queries are currently in `query-history-dto-mapper` and `query-history-domain-mapper` but will be moved into local queries specific files in a separate PR.

## Checklist
N/A:
- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/query-history/store/variant-analysis-dto-mapper.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
+import {
```
Comment (robertbrignull) [inline extensions/ql-vscode/src/query-history/store/variant-analysis-dto-mapper.ts:1]: Should this file be called `query-history-variant-analysis-dto-mapper.ts`, to fit with `query-history-variant-analysis-dto.ts`?

(Also applies to the domain mapper)

**Review by robertbrignull — COMMENTED** [review]
A couple of small questions, but looks pretty good to me

File: extensions/ql-vscode/src/query-history/store/query-history-domain-mapper.ts [hunk 2]
```diff
@@ -15,20 +15,19 @@ import {
   SortDirectionDto,
 } from "./query-history-local-query-dto";
 import { QueryHistoryItemDto } from "./query-history-dto";
-import { QueryHistoryVariantAnalysisDto } from "./query-history-variant-analysis-dto";
 import {
   RawResultsSortState,
   SortDirection,
   SortedResultSetInfo,
 } from "../../pure/interface-types";
+import { mapQueryHistoryVariantAnalysisToDto } from "./variant-analysis-domain-mapper";
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/src/query-history/store/query-history-domain-mapper.ts:23]: Yep, I'd like to pull the local stuff out as a next step. See PR description.

File: extensions/ql-vscode/src/query-history/store/variant-analysis-dto-mapper.ts (hunk 1 above)
Comment (charisk) [reply extensions/ql-vscode/src/query-history/store/variant-analysis-dto-mapper.ts:1]: Good shout.

File: extensions/ql-vscode/src/query-history/store/query-history-domain-mapper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (robertbrignull) [reply extensions/ql-vscode/src/query-history/store/query-history-domain-mapper.ts]: Oh sorry I missed that bit. Sounds good to me then.

**Review by robertbrignull — APPROVED** [review]
🚀

**Pull request merged**